### DISCUSSION TITLE
Max Message Size in RabbitMQ .NET client 6.4.0 up to 7.0.0

### DISCUSSION THREAD

**Discussion by dank100** [rabbitmq/rabbitmq-dotnet-client#1379] [General]
From 6.4.0 and forward There is now a limit on the message size, as you rightfully explain in your changelogs:

"This release adds the ability to specify a maximum message size when receiving data. The default values are:

RabbitMQ .NET client 7.0.0 and beyond: 128MiB

RabbitMQ .NET client 6.4.0 up to 7.0.0: no limit by default"

However, when I specify the cf.MaxMessageSize = 536870912 as in your example. I'm still limited by the max size of the message, i.e. 134217728.

If I downgrade to 6.3.1, I'm not limited by the MessageSize, on the very same cluster and exchange.

I'm not sure this is any bug, but I cannot seem to figure it out.

An observation I've made in ConnectionFactory.cs is that on your main branch we have. 

`public const uint DefaultMaxMessageSize = 134217728;`

 However, on, tag 6.5.0 we have.

`public const uint DefaultMaxMessageSize = 0;`

In your docs you specify that 0 means "unlimited".

With me using 6.5.0, I would expect the MaxMessageSize being 0, whether I specify it or not. However, as explained I'm limited by the MaxMessageSize of 134217728.

Is this anything you've heard of before, and are you able to pinpoint me in the right direction?

- Dan

**Comment by lukebakken** [comment]
Thanks for pointing this out, I have created the following issue to discuss further - https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1572

**Comment by michaelklishin** [comment]
@dank100 unlimited message size is a terrible idea. We have well over 10 years of practical experience. Anything beyond maybe 50-60 MiB belongs to a blob store.

"Unlimited message size" is not something that RabbitMQ supports, IIRC there is a hard cap at 512 MiB (it was last updated in January 2019 in https://github.com/rabbitmq/rabbitmq-common/pull/289/files).

Plus see https://github.com/rabbitmq/rabbitmq-server/issues/11187 to get the general idea of where the default (we can call it a "soft limit") is headed.
